docs/enhancements/tenancy-based-auth.md

```diff
@@ -0,0 +1,103 @@
+| List Cluster   | Returns clusters owned by all members of the organization                                      |
+| Get  Cluster   | Allowed for all members of the organization                                                    |
+| Edit Cluster   | Allowed for the owner and for each user that has update access for this cluster's subscription |
+| Delete Cluster | Allowed for the owner and for each user that has remove access for this cluster's subscription |
+
+For InfraEnv and InfraEnv-based resources like hosts (all APIs that starts with `/v2/infra-envs/{infra_env_id`):
+| Action         | Authorization Access                                                                           |
+|----------------|------------------------------------------------------------------------------------------------|
+| List InfraEnvs | Returns bound InfraEnvs belonging to the organization and unbound InfraEnvs owned by the caller                                                                                                            |
+| Get    InfraEnv| For bound InfraEnv, allowed for all members of the organization<br>For unbound InfraEnv allowed for the owner                                                                                                         |
+| Edit InfraEnv  | For bound InfraEnv, allowed for the owner and for each user that has update access<br>for the bound cluster's subscription. For unbound InfraEnv, allowed for the owner                                               |
+| Delete InfraEnv| For bound InfraEnv, allowed for the owner and for each user that has remove access<br>for the bound cluster's subscription. For unbound InfraEnv, allowed for the owner                                               |
+
+For day2 cluster operations:
+| Action         | Authorization Access                                                                           |
+|----------------|------------------------------------------------------------------------------------------------|
+| Add Host       | Allowed for every user who has update access for the day1's cluster subscription               |
+
+Filtering for you own clusters:
+The list-clusters API shall have an additional query parameter that filters the clusters by owners only:
+`GET /v2/clusters?owneronly=true`
+
+## Authorization Flow Overview
+![authorization flow](./auth_flow.png)
+
+Following is a brief overview of the authorization layer of the assisted installer and its interaction with AMS.
+
+There are three actors in the flow:
+- An authenticated client (User or Agent) carrying a bearer or access token that identifies it.
+- Assisted Service security middleware, acting as a reverse proxy, which receives a request to access a resource and works together with AMS to establish the right of the client to access it.
+- Account Management Service (AMS) that holds the OCM authorization details for a user.
```

Comment:
Is this a component that already exists? Is it part of OCM? Or is it being introduced in this enhancement?

Reply:
Part of OCM. perhaps the explanation of the acronyms at the beginning will clear the confusion but those acronyms are used sometime interchangeably in the openshift code as well